### PR TITLE
Add support for fig completions

### PR DESCRIPTION
**Pull request by crodas**
## Description

This fixes https://github.com/FuelLabs/sway/issues/5474

This is the first part of #5612, which will be split into several smaller PRs

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: forc-plugins/forc-crypto/src/main.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ mod keccak256;
 mod keys;
 mod sha256;
 
-const ABOUT: &str = "Forc plugin for hashing arbitrary data.";
+const ABOUT: &str = "Forc plugin for hashing arbitrary data";
```
Comment (sdankel) [inline forc-plugins/forc-crypto/src/main.rs:19]: Clap can also parse the About from a doc comment, so you don't have to pass it in.

```rust 
/// Forc plugin for hashing arbitrary data.
#[derive(Debug, Parser)]
#[clap(
    name = "forc-crypto",
    after_help = help(),
    version
)]
pub enum Command {
```

File: forc/src/cli/commands/completions.rs [hunk 2]
```diff
@@ -1,8 +1,36 @@
-use clap::Command as ClapCommand;
-use clap::{CommandFactory, Parser};
+use clap::{Command as ClapCommand, CommandFactory, Parser};
 use clap_complete::{generate, Generator, Shell};
 use forc_util::ForcResult;
 
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, clap::ValueEnum)]
+enum Target {
```
Comment (sdankel) [inline forc/src/cli/commands/completions.rs:6]: Same comment: https://github.com/FuelLabs/sway/pull/5612#discussion_r1493095479

File: forc/src/cli/commands/completions.rs [hunk 3]
```diff
@@ -1,8 +1,36 @@
...
+    /// Elvish shell
+    Elvish,
+    /// Friendly Interactive SHell (fish)
+    Fish,
+    /// PowerShell
+    PowerShell,
+    /// Z SHell (zsh)
+    Zsh,
+    /// Fig
+    Fig,
```
Comment (sdankel) [inline forc/src/cli/commands/completions.rs:18]: Fig is not a shell, right? It's an autocomplete provider for other shells.

**Review by sdankel — CHANGES_REQUESTED** [review]

**Review by sdankel — COMMENTED** [review]
Could you share the steps for manually testing this with fig?

Is it possible to write unit tests for this? It would be good to assert that the correct autocompletions file is created for each target.

**Review by JoshuaBatty — DISMISSED** [review]

**Review by kayagokalp — APPROVED** [review]
✅

**Review by JoshuaBatty — APPROVED** [review]

**Review by kayagokalp — APPROVED** [review]

**Pull request merged**